src/compiler/checker.ts

```diff
@@ -4414,7 +4413,7 @@ namespace ts {
          * boolean, and symbol primitive types, return the corresponding object types. Otherwise return the
          * type itself. Note that the apparent type of a union type is the union type itself.
          */
-        function getApparentType(type: Type): Type {
+        function getApparentType(type: Type): ObjectType {

```

Comment:
All the branches do return `ObjectType`, but I forgot about the else branch. Reverted.